src/charm.py

```diff
@@ -90,9 +94,13 @@ def _configure(self, event: EventBase) -> None:
             self.unit.status = WaitingStatus("Waiting for Multus to be ready")
             event.defer()
             return
+        if not self._n2_requirer.amf_hostname or not self._n2_requirer.amf_port:
```

Comment:
As it is the charm won't work if there is no relation with the fivegN2 so we should probably block when this relation is not set

Reply:
I added a check to see if the relation was created. However I think this check can stay too, for example if the `amf_ip` was not ready at the time of relation creation, `amf` might not update the relation data right away.